tests/testsuite.src/syn_occurs.at

```diff
@@ -501,3 +501,115 @@ AT_CHECK([$COMPILE_ONLY -std=mf -fodoslide prog.cob ], [0], [], [])
 
 AT_CLEANUP
 
+
+AT_SETUP([SET to out-of-range literal])
+AT_KEYWORDS([SET])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       01 G.
+         02 X           PIC X OCCURS 10 INDEXED I.
+       01 LVL-01-CST    CONSTANT 12.
+       78 LVL-78-CST    VALUE 13.
+       PROCEDURE        DIVISION.
+           SET I TO ZERO.
+           SET I TO 11.
+           SET I TO LVL-01-CST.
+           SET I TO LVL-78-CST.
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE_ONLY prog.cob], [0], [],
+[prog.cob:11: warning: SET I TO 0 is out of bounds
+prog.cob:12: warning: SET I TO 11 is out of bounds
+prog.cob:13: warning: SET I TO 12 is out of bounds
+prog.cob:14: warning: SET I TO 13 is out of bounds
+])
+
+AT_CLEANUP
+
+
+AT_SETUP([SET index])
+AT_KEYWORDS([SET])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01  BINB   PIC 9(9) COMP-5 VALUE 42.
+       01  NIDX   PIC S99.
+       01  MYIDX  USAGE IS INDEX.
+       01  MAXIDX PIC 9999 VALUE 3 COMP-5.
+       01  TBL.
+           05  FILLER PIC X(8) VALUE "Fred".
+           05  FILLER PIC X(8) VALUE "Barney".
+           05  FILLER PIC X(8) VALUE "Wilma".
+           05  FILLER PIC X(8) VALUE "Betty".
+       01  FILLER REDEFINES TBL.
+           05  MYNAME PIC X(8) OCCURS 4 INDEXED BY IB1.
+       01  TBL2.
+           05  MYMRK  PIC X(3)
+               OCCURS 2 TO 5 DEPENDING ON MAXIDX
+               INDEXED BY IB2.
+       PROCEDURE DIVISION.
+           MOVE 5 TO MAXIDX
+           SET NIDX TO IB1.
+           DISPLAY "Initial value: " NIDX.
+           SET IB2 TO 0.2.
+           SET IB2 TO "fred".
```

Comment:
this needs to be an error as well, then this roundup is complete

Reply:
The message `warning: numeric value is expected` for  `IB2 TO "fred"` comes from `validate_move`, itself called by `cb_build_move`, called from `cb_emit_set_to`. Guess I should not change that to a warning in `validate_move`. Should I just detect this case earlier in `cb_emit_set_to` to output a proper error ?

Reply:
yes, before, ideally in the parser already by using a different terminal which includes the check for numeric or integer constant

Reply:
I can't find any existing terminal that makes this check - should I create a new one ?

Reply:
Oh but actually, the `set_to` rule in `parser.y` seems to be used not only for indices (seems to be used at least for `ADDRESS OF` too). Seems the check must be in `typeck.c`.

Reply:
But both `set_to`  and `set_up_down` should not use `x` but a numeric or pointer or integer value (which possibly needs a new rule `x_numeric` or similar), which would cover all of this, so that `cb_emit_set_to` has only to check for the pointer part, no?

Reply:
Maybe, though, trying to add this check, this raised some questions.

1 / In `syn_misc.at` test "invalid use of condition-name", there is a SET to a level-88 condition, and the following expected output:
```
prog.cob:33: error: condition-name not allowed here: 'val-i1'
```
Since we're now forbiding anything but integer/pointer at parsing, the reported error will be different ; is that okay ?

2 / In `syn_move.at` test "MOVE FIGURATIVE to NUMERIC", we have `SET MYFLD TO HIGH-VALUES` and `SET MYFLD TO SPACES`, with an expected error of "error: invalid SET statement" - again this will change the error message : is it okay ?

Reply:
changing the error messages is fine
concerning 1: we can make that check as well "integer expected, not condition-name 'val-i1'"  or similar

Reply:
... but we may can drop the check "later on" that currently triggers these messages as those would be dead code then